lib/Alien/Build/Util.pm

```diff
@@ -76,7 +76,26 @@ sub _mirror
         { unlink "$dst" }
         my $target = readlink "$src";
         Alien::Build->log("ln -s $target $dst") if $opt->{verbose};
+        if (path($target)->is_relative) {
+          my $nativesymlink =
+                   (($^O eq "msys" && defined $ENV{MSYS} && $ENV{MSYS} eq "winsymlinks:nativestrict")
+                || ($^O eq "cygwin" && defined $ENV{CYGWIN} && $ENV{CYGWIN} eq "winsymlinks:nativestrict"));
+          if (!$src->parent->child($target)->exists) {
+            if ($nativesymlink) {
+              # NOTE: On linux, it is OK to create broken symlinks, but it is not allowed on 
+              #   windows MSYS2/Cygwin when nativestrict is used.
+              die "cannot create native symlink to nonexistent file $target on $^O";
+            }
+          }
+          if ($nativesymlink) {
+            $dst->parent->child($target)->touchpath;
+          }
+        }
+        my $curdir = Path::Tiny->cwd;
+        # CD into the directory, such that symlink will work on MSYS2
+        chdir $dst->parent or die "could not chdir to $src->parent : $!";
```

Comment:
I would like more illumination on why we are changing into the destination directory?

Reply:
For a relative target, the `symlink` call will fail if the target does not exist. The existence check is done relative to the current directory (tested this now).